com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/ProcessLoggerHandler.java

```diff
@@ -0,0 +1,155 @@
+    }
+
+    public ProcessLogger getLogger(DelegateExecution context, String namePrefix, String logName) {
+        return getLogger(context, namePrefix, logName, null);
+    }
+
+    public ProcessLogger getLogger(DelegateExecution context, String namePrefix, String logName, PatternLayout customLayout) {
+        String correlationId = getCorrelationId(context);
+        String currentActivity = getTaskId(context);
+        String name = getLoggerName(namePrefix, correlationId, logName, currentActivity);
+        ProcessLogger logger = getExistingLogger(name);
+        if (logger == null) {
+            logger = createProcessLogger(name, correlationId, currentActivity, logName, customLayout);
+            addLogger(name, logger);
+            return logger;
+        }
+        return logger;
+    }
+
+    private String getCorrelationId(DelegateExecution context) {
+        return (String) context.getVariable(Constants.CORRELATION_ID);
+    }
+
+    private String getTaskId(DelegateExecution context) {
+        String taskId = (String) context.getVariable(Constants.TASK_ID);
+        return taskId != null ? taskId : context.getCurrentActivityId();
+    }
+
+    private void appendLoggerContent(DelegateExecution context, ProcessLogger logger,
+        ProcessLogsPersistenceService processLogsPersistenceService) {
+        try {
+            logger.appendLogContent(context, processLogsPersistenceService);
+        } catch (FileStorageException e) {
+            LOGGER.warn(MessageFormat.format(Messages.COULD_NOT_PERSIST_LOGS_FILE, e.getMessage()), e);
+        } finally {
+            logger.deleteLogFile();
+        }
+    }
+
+    private String getLoggerName(String prefix, String logId, String name, String suffix) {
+        return new StringBuilder(prefix).append(".")
+            .append(logId)
+            .append(".")
+            .append(name)
+            .append(".")
+            .append(suffix)
+            .toString();
+    }
+
+    private ProcessLogger getExistingLogger(String name) {
+        return loggers.get(name);
+    }
+
+    private void addLogger(String name, ProcessLogger logger) {
+        loggers.put(name, logger);
+    }
+
+    private ProcessLogger createProcessLogger(String name, String processId, String activityId, String logName,
+        PatternLayout customLayout) {
+        Logger logger = createLogger(name);
+        File logFile = getLocalFile(processId, logName, activityId);
+        logger.addAppender(createAppender(logger.getLevel(), logFile, customLayout));
+        return new ProcessLogger(logger, logFile, logName, processId, activityId);
+    }
+
+    private Logger createLogger(String name) {
+        Logger logger;
+        logger = Logger.getLogger(name);
+        Level level = logger.getParent()
+            .getLevel();
+        logger.setLevel(level);
+        return logger;
+    }
+
+    private File getLocalFile(String logId, String name, String nameSuffix) {
```

Comment:
Why are you calling the processId "logId" and logName "name" here? I'm having a tough time following the flow of information through the methods.